src/tools/useLocalStorage.ts

```diff
@@ -45,8 +49,11 @@ export default function useLocalStorage(
   );
 
   useEffect(() => {
-    window.addEventListener("storage", handleStorage);
-    return () => window.removeEventListener("storage", handleStorage);
+    if (isWindowDefined) {
+      window.addEventListener("storage", handleStorage);
+      return () => window.removeEventListener("storage", handleStorage);
+    }
+    return;
```

Comment:
```suggestion

```

Reply:
tsconfig has "noImplicitReturns": true. I've changed it to:
```tsx
    isWindowDefined && window.addEventListener("storage", handleStorage);
    return () => {
      isWindowDefined && window.removeEventListener("storage", handleStorage);
    };
```
Is it ok or turn noImplicitReturns off?

Reply:
Looks ok to me

Reply:
Yes, tsc and linter shows it, the rule is about implicit/explicit return, not the return type

Reply:
Ok it will only throw if return and no-return are mixed together, in that case let's go back to the original solution 👍